samtranslator/model/eventsources/pull.py

```diff
@@ -355,9 +355,46 @@ class MQ(PullEventSource):
     property_types: Dict[str, PropertyType] = {
         **PullEventSource.property_types,
         "Broker": PassThroughProperty(True),
+        "DynamicPolicyName": Property(False, is_type(bool)),
     }
 
     Broker: PassThrough
+    DynamicPolicyName: Optional[bool]
+
+    @property
+    def policy_name(self) -> str:
```

Comment:
nit: rename to `_policy_name` so less likely be mistaken as a public function

Reply:
That private access in python is just an illusion anyway and I'm not fond of us using them, but why should it be private? it's a read only property, what's wrong with it being public?

Reply:
It's to minimize public (by convention, yes there's no "real" private) API footprint so that we have the freedom to change things in the future. We've had instances of e.g. SAM CLI depending on functions in our codebase that we changed, causing errors. And since we never publicized a library-level backward compatibility policy, we err toward being cautious and assuming everything by convention public is depended on.

Reply:
Sure,  I'll change ig